### ISSUE TITLE
LCP PerformanceEntry should Extend from PerformanceElementTiming entry

### ISSUE THREAD

**Issue by mmocny** [yoavweiss/LargestContentfulPaint#127]
LCP entries expose a `size` but not an [`intersectionRect`](https://wicg.github.io/element-timing/#dom-performanceelementtiming-intersectionrect).

Is there a specific reason?

Having the full rect is quite useful especially for cases such as:

* the `element` becomes `null` for reporting
* the sizing of the element is constrained to a bounding box that does not match `getBoundingClientRect()`
  * (and as far as I know getting a tight bounding box is really hard and involves manually parsing computed styles)
* Useful to visualize cases when surprising candidates or sizes are computed (i.e. when there are nested inline text nodes, etc).

**Comment by mmocny** [comment]
Looking at the algorithm for LCP, it is explicitly taking an `intersectionRect` as computed by element timing and then converting just to a single size:

https://w3c.github.io/largest-contentful-paint/#potentially-add-a-largestcontentfulpaint-entry

**Comment by mmocny** [comment]
...meanwhile element timing does not expose `size`.

I think these entry types should try to be as similar as possible, no?  Then you can share a sinlge PerformanceObserver, for example.

**Comment by clelland** [comment]
I agree; this would have been really useful in a number of debugging situations. I don't know if this was an oversight in the original design or not, but it would be a useful addition now.

**Comment by bdekoz** [comment]
agree. Moz has an existing use case for Michal's third point above, aka "Useful to visualize cases when surprising candidates.."

**Comment by mmocny** [comment]
Excellent!  I'll ask if we can add to the WG meeting schedule, just to confirm this is not intentional for some obscure historical reason.

Should be trivial to patch into spec.

**Comment by mmocny** [comment]
One small caveat: @clelland points out that the intersectionRect size might not be == to the size property whenever LCP is capped in size based on intrinsic size of media.  (I think thats fine, and actually a useful hint to know when LCP is affected by this)

**Comment by mmocny** [comment]
[Here is a snippet](https://gist.github.com/mmocny/3c47fcfb75c825d801516bd85faf7f29) that you can use, if you have intersectionObserver, to visualize these rects (Thanks @philipwalton).

**Comment by mmocny** [comment]
Discussed at the last Web Perf WG and there are no concerns and general agreement about adding both `intersectionRect` to LCP and `size` to element-timing.

There was some feedback about if that is _enough_ and I'm going to follow up to see if I can articulate the requirements and post back here.

(We also discussed issues related to element timing API and scrolling and viewport intersections -- but these are more about timing than the properties being exposed)

**Comment by mmocny** [comment]
...oh and we also discussed that the Interop 2025 proposal would NOT include any changes to WPT by default, but there is a pathway towards making a request to add.  If we update spec and WPT this quarter there were no concerns from implementers in the room about trying to do that-- but its also not critical.

**Comment by mmocny** [comment]
(Updated issue title based on this context: https://github.com/w3c/largest-contentful-paint/issues/133#issuecomment-3168917747)
